subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/PrecompiledScriptPluginIntegrationTest.kt

```diff
@@ -550,4 +551,66 @@ class PrecompiledScriptPluginIntegrationTest : AbstractPluginIntegrationTest() {
     private
     fun CharSequence.count(text: CharSequence): Int =
         StringGroovyMethods.count(this, text)
+
+    // https://github.com/gradle/gradle/issues/15416
+    @Ignore // not yet implemented
+    @Test
+    @ToBeFixedForConfigurationCache(because = "Kotlin Gradle Plugin")
+    fun `can use plugins block in precompiled settings plugin`() {
+        withFolders {
+            "build-logic" {
+                withFile(
+                    "settings.gradle.kts",
+                    """
+                        $defaultSettingsScript
+                    """
+                )
+                withFile(
+                    "build.gradle.kts",
+                    """
+                        plugins {
+                            `kotlin-dsl`
+                        }
+                        $repositoriesBlock
+                    """
+                )
+                withFile(
+                    "src/main/kotlin/base-plugin.settings.gradle.kts",
+                    """
+                        println("base-plugin settings plugin applied")
+                    """
+                )
+                withFile(
+                    "src/main/kotlin/my-plugin.settings.gradle.kts",
+                    """
+                        plugins {
+                            id("base-plugin")
```

Comment:
We could break this down into two or three cases:
1. Just having an empty plugin block
1. Applying an existing plugin (e.g. enterprise plugin or a plugin from a custom repo in the test folder)
2. What you have now: Apply a plugin defined in the same project